pkg/cluster/fetch.go

```diff
@@ -211,7 +262,21 @@ func BuildSpec(ctx context.Context, client Client, cluster *v1alpha1.Cluster) (*
 
 // BuildSpecFromConfig constructs a cluster.Spec for an eks-a cluster config by retrieving all dependencies objects from the cluster using a kubernetes client.
 func BuildSpecFromConfig(ctx context.Context, client Client, config *Config) (*Spec, error) {
-	bundlesName, bundlesNamespace := bundlesNamespacedKey(config.Cluster)
+	eksaRelease := &v1alpha1release.EKSARelease{}
+	if config.Cluster.Spec.BundlesRef == nil {
+		if config.Cluster.Spec.EksaVersion == nil {
+			return nil, fmt.Errorf("cluster's EksaVersion cannot be nil")
+		}
+		version := string(*config.Cluster.Spec.EksaVersion)
+		eksaReleaseName := v1alpha1release.GenerateEKSAReleaseName(version)
+		if err := client.Get(ctx, eksaReleaseName, constants.EksaSystemNamespace, eksaRelease); err != nil {
+			return nil, fmt.Errorf("error getting EKSARelease %s", eksaReleaseName)
+		}
+	} else {
+		eksaRelease = nil
+	}
+
+	bundlesName, bundlesNamespace := bundlesNamespacedKey(config.Cluster, eksaRelease)
```

Comment:
how does this work if the eksaRelease is nil?

Reply:
eksaRelease isn't needed in bundlesNamespacedKey, but I did remove setting it to nil
bundlesNamespacedKey will fall back to the cluster's BundlesRef if the eksaRelease's BundlesRef is empty